node/src/BaseClient.ts

```diff
@@ -348,20 +348,63 @@ function convertGlideRecordForZSet(
 
 /**
  * @internal
- * Downcast `GlideRecord` to `Record`. Use if you are 146% aware that `data` keys are always strings.
+ * Recursevly downcast `GlideRecord` to `Record`. Use if `data` keys are always strings.
```

Comment:
That will crash if a key is a `Buffer`. I use it in tests and while converting data for streams (stream IDs are strings)

Reply:
Oh - because it's not safe. 
```
Only use if the key attribute in `data` will always be if string type.  
```
Maybe specifically indicate that it should only be used for stream data containing stream ids?